client/post-editor/controller.js

```diff
@@ -199,5 +200,30 @@ module.exports = {
 
 		page.redirect( redirectWithParams );
 		return false;
-	}
+	},
+
+	copyPost: function( context, next ) {
+		if ( ! context.query.copy ) {
+			// not copyPost, early return
+			return next();
+		}
+
+		const postId = getPostID( context );
+
+		function startEditingOnSiteSelected() {
+			const siteId = getSelectedSiteId( context.store.getState() );
+			if ( siteId ) {
+				startEditingPostCopy( context, siteId, postId ).then( () => {
+					renderEditor( context, 'post' );
+				} );
+			} else {
+				sites.once( 'change', startEditingOnSiteSelected );
+			}
+		}
+
+		startEditingOnSiteSelected();
+
+		return false;
```

Comment:
I might have expected the middleware to still leverage the `post` function in calling `next`, perhaps by setting the loaded post values into `context` and detecting those in the `post` function. We should also consider rendering the editor before sites or the post have finished loading as we do in the `post` function.